--- CMakeLists.txt-old	2012-09-22 20:54:07.212455303 -0500
+++ CMakeLists.txt	2012-09-22 20:54:49.051697427 -0500
@@ -209,17 +209,6 @@
    PATTERN "*/.svn*" EXCLUDE
    PATTERN "*/.git*" EXCLUDE
    PATTERN "*~" EXCLUDE)
-install(DIRECTORY unittests DESTINATION share/chaiscript
-   PATTERN "*.chai" 
-   PATTERN "*.inc"
-   PATTERN "*/.svn*" EXCLUDE
-   PATTERN "*/.git*" EXCLUDE
-   PATTERN "*~" EXCLUDE)
-install(DIRECTORY samples DESTINATION share/chaiscript
-   PATTERN "*.chai"
-   PATTERN "*/.svn*" EXCLUDE
-   PATTERN "*/.git*" EXCLUDE
-   PATTERN "*~" EXCLUDE)
 
 configure_file(contrib/pkgconfig/chaiscript.pc.in lib/pkgconfig/chaiscript.pc @ONLY)
 install(FILES "${chaiscript_BINARY_DIR}/lib/pkgconfig/chaiscript.pc"
