From 05e338730b3dc8aef8e433b7c8e32b43815565fa Mon Sep 17 00:00:00 2001
From: Matt Whitlock <c-lightning@mattwhitlock.name>
Date: Sat, 13 Feb 2021 15:34:52 -0500
Subject: [PATCH] libwally-core 0.8.2 compatibility

---
 common/setup.c | 3 +++
 configure      | 7 +++++++
 2 files changed, 10 insertions(+)

diff --git a/common/setup.c b/common/setup.c
index 2ea41f5a2..65e3d9011 100644
--- a/common/setup.c
+++ b/common/setup.c
@@ -19,6 +19,9 @@ static void wally_free(void *ptr)
 }
 
 static struct wally_operations wally_tal_ops = {
+#if HAVE_WALLY_OPERATIONS_STRUCT_SIZE
+	.struct_size = sizeof wally_tal_ops,
+#endif
 	.malloc_fn = wally_tal,
 	.free_fn = wally_free,
 };
diff --git a/configure b/configure
index b6b1e2cd3..211a0d425 100755
--- a/configure
+++ b/configure
@@ -375,6 +375,13 @@ int main(void)
 	return 1;
 }
 /*END*/
+var=HAVE_WALLY_OPERATIONS_STRUCT_SIZE
+desc=struct_size in struct wally_operations
+style=OUTSIDE_MAIN
+code=
+#include <libwally/wally_core.h>
+static struct wally_operations ops = { .struct_size = sizeof ops };
+/*END*/
 EOF
 
 if check_command 'python3-mako' python3 -c 'import mako'; then
-- 
2.30.1

