diff -Naur a/Makefile b/Makefile
--- a/Makefile	2020-08-28 06:33:06.311457074 +0000
+++ b/Makefile	2020-08-28 06:38:26.295542542 +0000
@@ -228,7 +228,7 @@
 
 # If we have the postgres client library we need to link against it as well
 ifeq ($(HAVE_POSTGRES),1)
-LDLIBS += -lpq
+LDLIBS += $(POSTGRES_LDLIBS)
 endif
 
 default: show-flags all-programs all-test-programs doc-all
diff -Naur a/configure b/configure
--- a/configure	2020-08-28 06:33:06.689460719 +0000
+++ b/configure	2020-08-28 06:40:47.537904476 +0000
@@ -230,8 +230,10 @@
 require 'python3-mako' "You need the mako module for python3: see doc/INSTALL.md" python3 -c 'import mako'
 
 POSTGRES_INCLUDE=""
-if command -v pg_config 2> /dev/null; then
-    POSTGRES_INCLUDE="-I$(pg_config --includedir)"
+POSTGRES_LDLIBS=""
+if command -v "${PG_CONFIG-pg_config}" 2> /dev/null; then
+    POSTGRES_INCLUDE="-I$("${PG_CONFIG-pg_config}" --includedir)"
+    POSTGRES_LDLIBS="-L$("${PG_CONFIG-pg_config}" --libdir) -lpq"
 fi
 
 rm -f $CONFIG_VAR_FILE.$$
@@ -290,7 +292,7 @@
 var=HAVE_POSTGRES
 desc=postgres
 style=DEFINES_EVERYTHING|EXECUTE|MAY_NOT_COMPILE
-link=-lpq
+link=${POSTGRES_LDLIBS}
 code=
 #include <libpq-fe.h>
 #include <stdio.h>
@@ -360,6 +362,7 @@
 add_var CDEBUGFLAGS "$CDEBUGFLAGS"
 add_var COPTFLAGS "$COPTFLAGS"
 add_var POSTGRES_INCLUDE "$POSTGRES_INCLUDE"
+add_var POSTGRES_LDLIBS "$POSTGRES_LDLIBS"
 add_var VALGRIND "$VALGRIND"
 add_var DEVELOPER "$DEVELOPER" $CONFIG_HEADER
 add_var EXPERIMENTAL_FEATURES "$EXPERIMENTAL_FEATURES" $CONFIG_HEADER
