From f09ced2747e25d495b29be87189fa816dfc97ca8 Mon Sep 17 00:00:00 2001
From: Matt Whitlock <c-lightning@mattwhitlock.name>
Date: Sat, 8 Oct 2022 22:07:13 -0400
Subject: [PATCH 2/2] swig_python: link dynamically to libwallycore.so

---
 setup.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 9de4163..284e2a1 100644
--- a/setup.py
+++ b/setup.py
@@ -79,12 +79,11 @@ wally_ext = Extension(
     '_wallycore',
     define_macros=define_macros,
     include_dirs=include_dirs,
+    library_dirs=['src/.libs'],
+    libraries=['wallycore'],
     extra_compile_args=extra_compile_args,
     sources=[
         'src/swig_python/swig_wrap.c' if is_windows else 'src/swig_python/swig_python_wrap.c',
-        'src/amalgamation/combined.c',
-        'src/amalgamation/combined_ccan.c',
-        'src/amalgamation/combined_ccan2.c',
         ],
     )
 pkgconfig.configure_extension(wally_ext, 'libsecp256k1_zkp')
-- 
2.41.0

From 245f1c1a9e300e0281702ee37f6a5632ac648beb Mon Sep 17 00:00:00 2001
From: Matt Whitlock <c-lightning@mattwhitlock.name>
Date: Tue, 6 Jun 2023 00:05:09 -0400
Subject: [PATCH] swig_python: add -Wno-unused-parameter to SWIG_WARN_CFLAGS

SWIG generates code with unused parameters that trigger hundreds of warnings.
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 1068f4d..458a44e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -180,6 +180,7 @@ AC_SUBST([NOBUILTIN_CFLAGS])
 
 # SWIG versions vary in generated code quality; skip warnings
 SWIG_WARN_CFLAGS="-fno-strict-aliasing"
+AX_CHECK_COMPILE_FLAG([-Wno-unused-parameter], [SWIG_WARN_CFLAGS="$SWIG_WARN_CFLAGS -Wno-unused-parameter"])
 AX_CHECK_COMPILE_FLAG([-Wno-shadow], [SWIG_WARN_CFLAGS="$SWIG_WARN_CFLAGS -Wno-shadow"])
 AX_CHECK_COMPILE_FLAG([-Wno-self-assign], [SWIG_WARN_CFLAGS="$SWIG_WARN_CFLAGS -Wno-self-assign"])
 AX_CHECK_COMPILE_FLAG([-Wno-missing-field-initializers], [SWIG_WARN_CFLAGS="$SWIG_WARN_CFLAGS -Wno-missing-field-initializers"])
-- 
2.41.0

