--- /var/cache/distfiles/31732f7825ab59878cb4704bc2a7b5384cf9c2a0.patch	2022-12-21 21:17:08.092350752 -0500
+++ /var/tmp/portage/net-p2p/core-lightning-0.11.2-r112/temp/31732f7825ab59878cb4704bc2a7b5384cf9c2a0.patch	2022-12-21 22:00:23.167788537 -0500
@@ -236,8 +237,8 @@
  [comment]: # (GENERATE-FROM-SCHEMA-START)
  On success, an object is returned, containing:
  
--- **type** (string): what kind of object it decoded to (one of "bolt12 offer", "bolt12 invoice", "bolt12 invoice_request", "bolt11 invoice", "rune")
-+- **type** (string): what kind of object it decoded to (one of "bolt12 offer", "bolt12 invoice", "bolt12 invoice\_request", "bolt11 invoice", "rune")
+-- **type** (string): what kind of object it decoded to (one of "bolt12 offer", "bolt12 invoice", "bolt12 invoice_request", "bolt11 invoice")
++- **type** (string): what kind of object it decoded to (one of "bolt12 offer", "bolt12 invoice", "bolt12 invoice\_request", "bolt11 invoice")
  - **valid** (boolean): if this is false, you *MUST* not use the result except for diagnostics!
  
  If **type** is "bolt12 offer", and **valid** is *true*:
@@ -461,7 +462,7 @@
 index 4568ebfbb9c02eb5c0aaa451f9812aab8c28da46..9414040d5763ccccf13b767cfdbb85dc97a16192 100644
 --- a/doc/lightning-delpay.7.md
 +++ b/doc/lightning-delpay.7.md
-@@ -42,7 +42,7 @@ payments will be returned -- one payment object for each partid.
+@@ -42,14 +42,14 @@ payments will be returned -- one payment object for each partid.
  On success, an object containing **payments** is returned.  It is an array of objects, where each object contains:
  
  - **id** (u64): unique ID for this payment attempt
@@ -470,10 +471,9 @@
  - **status** (string): status of the payment (one of "pending", "failed", "complete")
  - **amount\_sent\_msat** (msat): the amount we actually sent, including fees
  - **created\_at** (u64): the UNIX timestamp showing when this payment was initiated
-@@ -50,7 +50,7 @@ On success, an object containing **payments** is returned.  It is an array of ob
+ - **partid** (u64, optional): unique ID within this (multi-part) payment
  - **destination** (pubkey, optional): the final destination of the payment if known
  - **amount\_msat** (msat, optional): the amount the destination received, if known
- - **completed\_at** (u64, optional): the UNIX timestamp showing when this payment was completed
 -- **groupid** (u64, optional): Grouping key to disambiguate multiple attempts to pay an invoice or the same payment_hash
 +- **groupid** (u64, optional): Grouping key to disambiguate multiple attempts to pay an invoice or the same payment\_hash
  - **payment\_preimage** (hex, optional): proof of payment (always 64 characters)
@@ -821,9 +831,9 @@
 --- a/doc/lightning-listconfigs.7.md
 +++ b/doc/lightning-listconfigs.7.md
 @@ -101,7 +101,7 @@ On success, an object is returned, containing:
+ - **fetchinvoice-noconnect** (boolean, optional): `fetchinvoice-noconnect` fields from config or cmdline, or default
  - **accept-htlc-tlv-types** (string, optional): `accept-extra-tlvs-type` fields from config or cmdline, or not present
  - **tor-service-password** (string, optional): `tor-service-password` field from config or cmdline, if any
- - **dev-allowdustreserve** (boolean, optional): Whether we allow setting dust reserves
 -- **announce-addr-dns** (boolean, optional): Whether we put DNS entries into node_announcement
 +- **announce-addr-dns** (boolean, optional): Whether we put DNS entries into node\_announcement
  
@@ -849,19 +859,16 @@
 index 88570c472e2775b89e64ed8878113153fc71a9bc..7fa489a96bac5695651ba380ae247971323c901f 100644
 --- a/doc/lightning-listforwards.7.md
 +++ b/doc/lightning-listforwards.7.md
-@@ -26,23 +26,23 @@ On success, an object containing **forwards** is returned.  It is an array of ob
- 
+@@ -27,7 +27,7 @@ On success, an object containing **forwards** is returned.  It is an array of ob
  - **in\_channel** (short\_channel\_id): the channel that received the HTLC
+ - **in\_htlc\_id** (u64): the unique HTLC id the sender gave this
  - **in\_msat** (msat): the value of the incoming HTLC
 -- **status** (string): still ongoing, completed, failed locally, or failed after forwarding (one of "offered", "settled", "local_failed", "failed")
 +- **status** (string): still ongoing, completed, failed locally, or failed after forwarding (one of "offered", "settled", "local\_failed", "failed")
  - **received\_time** (number): the UNIX timestamp when this was received
- - **in\_htlc\_id** (u64, optional): the unique HTLC id the sender gave this (not present if incoming channel was closed before ugprade to v22.11)
  - **out\_channel** (short\_channel\_id, optional): the channel that the HTLC (trying to) forward to
--- **out\_htlc\_id** (u64, optional): the unique HTLC id we gave this when sending (may be missing even if out_channel is present, for old forwards before v22.11)
-+- **out\_htlc\_id** (u64, optional): the unique HTLC id we gave this when sending (may be missing even if out\_channel is present, for old forwards before v22.11)
- - **style** (string, optional): Either a legacy onion format or a modern tlv format (one of "legacy", "tlv")
- 
+ - **out\_htlc\_id** (u64, optional): the unique HTLC id we gave this when sending
+@@ -36,13 +36,13 @@ On success, an object containing **forwards** is returned.  It is an array of ob
  If **out\_msat** is present:
  
    - **fee\_msat** (msat): the amount this paid in fees
@@ -1015,8 +1022,8 @@
 +  - **state** (string): the channel state, in particular "CHANNELD\_NORMAL" means the channel can be used normally (one of "OPENINGD", "CHANNELD\_AWAITING\_LOCKIN", "CHANNELD\_NORMAL", "CHANNELD\_SHUTTING\_DOWN", "CLOSINGD\_SIGEXCHANGE", "CLOSINGD\_COMPLETE", "AWAITING\_UNILATERAL", "FUNDING\_SPEND\_SEEN", "ONCHAIN", "DUALOPEND\_OPEN\_INIT", "DUALOPEND\_AWAITING\_LOCKIN")
    - **opener** (string): Who initiated the channel (one of "local", "remote")
    - **features** (array of strings):
--    - BOLT #9 features which apply to this channel (one of "option_static_remotekey", "option_anchor_outputs", "option_zeroconf")
-+    - BOLT #9 features which apply to this channel (one of "option\_static\_remotekey", "option\_anchor\_outputs", "option\_zeroconf")
+-    - BOLT #9 features which apply to this channel (one of "option_static_remotekey", "option_anchor_outputs")
++    - BOLT #9 features which apply to this channel (one of "option\_static\_remotekey", "option\_anchor\_outputs")
    - **scratch\_txid** (txid, optional): The txid we would use if we went onchain now
    - **feerate** (object, optional): Feerates for the current tx:
      - **perkw** (u32): Feerate per 1000 weight (i.e kSipa)
@@ -1030,7 +1037,7 @@
    - **funding\_outnum** (u32, optional): The 0-based output number of the funding transaction which opens the channel
    - **initial\_feerate** (string, optional): For inflight opens, the first feerate used to initiate the channel open
 @@ -102,8 +102,8 @@ On success, an object containing **peers** is returned.  It is an array of objec
-     - **remote** (short\_channel\_id, optional): An alias assigned by the remote node to this channel, usable in routehints and invoices
+   - **max\_accepted\_htlcs** (u32, optional): Maximum number of incoming HTLC we will accept at once
    - **state\_changes** (array of objects, optional): Prior state changes:
      - **timestamp** (string): UTC timestamp of form YYYY-mm-ddTHH:MM:SS.%03dZ
 -    - **old\_state** (string): Previous state (one of "OPENINGD", "CHANNELD_AWAITING_LOCKIN", "CHANNELD_NORMAL", "CHANNELD_SHUTTING_DOWN", "CLOSINGD_SIGEXCHANGE", "CLOSINGD_COMPLETE", "AWAITING_UNILATERAL", "FUNDING_SPEND_SEEN", "ONCHAIN", "DUALOPEND_OPEN_INIT", "DUALOPEND_AWAITING_LOCKIN")
@@ -1492,7 +1499,7 @@
 +- **groupid** (u64, optional): Grouping key to disambiguate multiple attempts to pay an invoice or the same payment\_hash
  - **amount\_msat** (msat, optional): The amount delivered to destination (if known)
  - **destination** (pubkey, optional): the final destination of the payment if known
- - **completed\_at** (u64, optional): the UNIX timestamp showing when this payment was completed
+ - **label** (string, optional): the *label*, if given to sendpay
 @@ -84,7 +84,7 @@ On success, an object is returned, containing:
  
  If **status** is "complete":
@@ -1767,7 +1774,7 @@
 +- **groupid** (u64, optional): Grouping key to disambiguate multiple attempts to pay an invoice or the same payment\_hash
  - **amount\_msat** (msat, optional): The amount delivered to destination (if known)
  - **destination** (pubkey, optional): the final destination of the payment if known
- - **completed\_at** (number, optional): the UNIX timestamp showing when this payment was completed
+ - **label** (string, optional): the label, if given to sendpay
 @@ -51,7 +51,7 @@ On success, an object is returned, containing:
  
  If **status** is "complete":
