From ee6d95577363488c6d06dc81b35fd8e216ac15e5 Mon Sep 17 00:00:00 2001
From: Luke Dashjr <luke-jr+git@utopios.org>
Date: Thu, 11 Aug 2011 21:00:47 -0400
Subject: [PATCH] Backport reduced fees of 0.0005 BTC (send/accept) and 0.0001 BTC (relay)

---
 src/main.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.h b/src/main.h
index 117a084..7e0a47a 100644
--- a/src/main.h
+++ b/src/main.h
@@ -29,8 +29,8 @@ static const unsigned int MAX_BLOCK_SIZE_GEN = MAX_BLOCK_SIZE/2;
 static const int MAX_BLOCK_SIGOPS = MAX_BLOCK_SIZE/50;
 static const int64 COIN = 100000000;
 static const int64 CENT = 1000000;
-static const int64 MIN_TX_FEE = CENT;
-static const int64 MIN_RELAY_TX_FEE = 50000;
+static const int64 MIN_TX_FEE = 50000;
+static const int64 MIN_RELAY_TX_FEE = 10000;
 static const int64 MAX_MONEY = 21000000 * COIN;
 inline bool MoneyRange(int64 nValue) { return (nValue >= 0 && nValue <= MAX_MONEY); }
 static const int COINBASE_MATURITY = 100;
-- 
1.7.3.4

