From 4980404ff894cc211ca08c31d3a7e42175a0cfaa Mon Sep 17 00:00:00 2001
From: Luke Dashjr <luke-jr+git@utopios.org>
Date: Thu, 11 Aug 2011 20:48:28 -0400
Subject: [PATCH] Backport reduced base fee of 0.0005 BTC

---
 main.h |   15 +++++++++------
 1 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/main.h b/main.h
index 088e286..969e1f7 100644
--- a/main.h
+++ b/main.h
@@ -19,6 +19,7 @@ static const unsigned int MAX_BLOCK_SIZE_GEN = MAX_BLOCK_SIZE/2;
 static const int MAX_BLOCK_SIGOPS = MAX_BLOCK_SIZE/50;
 static const int64 COIN = 100000000;
 static const int64 CENT = 1000000;
+static const int64 MIN_TX_FEE = 50000;
 static const int64 MAX_MONEY = 21000000 * COIN;
 inline bool MoneyRange(int64 nValue) { return (nValue >= 0 && nValue <= MAX_MONEY); }
 static const int COINBASE_MATURITY = 100;
@@ -575,10 +576,12 @@ public:
 
     int64 GetMinFee(unsigned int nBlockSize=1, bool fAllowFree=true) const
     {
-        // Base fee is 1 cent per kilobyte
+        // Base fee is MIN_TX_FEE per kilobyte
+        int64 nBaseFee = MIN_TX_FEE;
+
         unsigned int nBytes = ::GetSerializeSize(*this, SER_NETWORK);
         unsigned int nNewBlockSize = nBlockSize + nBytes;
-        int64 nMinFee = (1 + (int64)nBytes / 1000) * CENT;
+        int64 nMinFee = (1 + (int64)nBytes / 1000) * nBaseFee;
 
         if (fAllowFree)
         {
@@ -597,11 +600,11 @@ public:
             }
         }
 
-        // To limit dust spam, require a 0.01 fee if any output is less than 0.01
-        if (nMinFee < CENT)
+        // To limit dust spam, require MIN_TX_FEE if any output is less than 0.01
+        if (nMinFee < nBaseFee)
             foreach(const CTxOut& txout, vout)
-                if (txout.nValue < CENT)
-                    nMinFee = CENT;
+                if (txout.nValue < nBaseFee)
+                    nMinFee = nBaseFee;
 
         // Raise the price as the block approaches full
         if (nBlockSize != 1 && nNewBlockSize >= MAX_BLOCK_SIZE_GEN/2)
-- 
1.7.3.4

