diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3fd1dd6..6878946 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,13 +1,15 @@
 cmake_minimum_required(VERSION 3.1)
 project(ur VERSION 0.1.0)
 
+include(GNUInstallDirs)
+
 option(UR__DISABLE_TESTS "If tests should be compiled or not" OFF)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-add_library("${PROJECT_NAME}" STATIC 
+add_library("${PROJECT_NAME}"
     src/bc32.cpp 
     src/ur.cpp 
     src/utils.cpp
@@ -19,7 +21,8 @@ if(WIN32)
 endif()
 
 target_include_directories("${PROJECT_NAME}" PUBLIC 
-    "${PROJECT_SOURCE_DIR}/src"
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>
 )
 
 if (NOT UR__DISABLE_TESTS)
@@ -30,3 +33,8 @@ endif()
 set(CPACK_PROJECT_NAME ${PROJECT_NAME})
 set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})
 include(CPack)
+
+set(${PROJECT_NAME}_HEADERS src/bc32.h src/ur.h src/utils.h)
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${${PROJECT_NAME}_HEADERS}")
+install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME} PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
+install(EXPORT ${PROJECT_NAME} DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
