diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64d7995..20eab33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.16)
 
 project(tap-protocol LANGUAGES CXX)
 
+include(GNUInstallDirs)
+
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -50,8 +52,10 @@ target_link_libraries(${PROJECT_NAME}
         )
 
 target_include_directories(${PROJECT_NAME}
-        PUBLIC ${PROJECT_SOURCE_DIR}/include
-        PUBLIC ${PROJECT_SOURCE_DIR}/contrib/include
+        PUBLIC
+        	$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+        	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+        PRIVATE ${PROJECT_SOURCE_DIR}/contrib/include
         )
 
 target_compile_definitions(${PROJECT_NAME} PRIVATE TAPPROTOCOL_LIBRARY)
@@ -101,3 +105,7 @@ if (BUILD_TESTING)
         add_test(NAME ${testcase} COMMAND ${testcase})
     endforeach ()
 endif ()
+
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${${PROJECT_NAME}_HEADERS}")
+install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME} PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/tap_protocol)
+install(EXPORT ${PROJECT_NAME} DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
diff --git a/contrib/bitcoin-core/CMakeLists.txt b/contrib/bitcoin-core/CMakeLists.txt
index 81b54bf..237f609 100644
--- a/contrib/bitcoin-core/CMakeLists.txt
+++ b/contrib/bitcoin-core/CMakeLists.txt
@@ -84,14 +84,9 @@ add_library(${PROJECT_NAME} STATIC
 if (TARGET secp256k1)
     target_link_libraries(${PROJECT_NAME} PUBLIC secp256k1)
 else()
-    set(SECP256K1_DISABLE_SHARED ON)
-    set(SECP256K1_ENABLE_MODULE_RECOVERY ON)
-    set(SECP256K1_BUILD_BENCHMARK OFF)
-    set(SECP256K1_BUILD_TESTS OFF)
-    set(SECP256K1_BUILD_EXHAUSTIVE_TESTS OFF)
-    set(SECP256K1_BUILD_CTIME_TESTS OFF)
-    set(SECP256K1_BUILD_EXAMPLES OFF)
-    add_subdirectory(src/secp256k1)
+    find_package(PkgConfig)
+    pkg_check_modules(LIBSECP256K1 REQUIRED IMPORTED_TARGET libsecp256k1)
+    add_library(secp256k1 ALIAS PkgConfig::LIBSECP256K1)
     target_link_libraries(${PROJECT_NAME} PUBLIC secp256k1)
 endif()
 
diff --git a/include/tap_protocol/cktapcard.h b/include/tap_protocol/cktapcard.h
index 89a7370..62efe05 100644
--- a/include/tap_protocol/cktapcard.h
+++ b/include/tap_protocol/cktapcard.h
@@ -4,7 +4,7 @@
 #include <memory>
 #include <optional>
 #include <vector>
-#include "nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 #include "transport.h"
 
 // NFCISO7816Tag for iOS
diff --git a/include/tap_protocol/hash_utils.h b/include/tap_protocol/hash_utils.h
index a3d4351..e534d67 100644
--- a/include/tap_protocol/hash_utils.h
+++ b/include/tap_protocol/hash_utils.h
@@ -1,7 +1,7 @@
 #ifndef HASH_UTILS_H
 #define HASH_UTILS_H
 
-#include "tap_protocol/tap_protocol.h"
+#include "tap_protocol.h"
 
 namespace tap_protocol {
 
diff --git a/include/tap_protocol/hwi_tapsigner.h b/include/tap_protocol/hwi_tapsigner.h
index 3377006..c55047c 100644
--- a/include/tap_protocol/hwi_tapsigner.h
+++ b/include/tap_protocol/hwi_tapsigner.h
@@ -4,8 +4,8 @@
 #include <memory>
 #include <optional>
 #include <string>
-#include "tap_protocol/cktapcard.h"
-#include "tap_protocol/tap_protocol.h"
+#include "cktapcard.h"
+#include "tap_protocol.h"
 
 namespace bc_core {
   struct CExtPubKey;
diff --git a/include/tap_protocol/secp256k1_utils.h b/include/tap_protocol/secp256k1_utils.h
index 609f7a2..2e1de36 100644
--- a/include/tap_protocol/secp256k1_utils.h
+++ b/include/tap_protocol/secp256k1_utils.h
@@ -1,7 +1,7 @@
 #ifndef SECP256K1_UTILS_H
 #define SECP256K1_UTILS_H
 
-#include "tap_protocol/tap_protocol.h"
+#include "tap_protocol.h"
 
 namespace tap_protocol {
 std::pair<Bytes, Bytes> CT_pick_keypair();
diff --git a/include/tap_protocol/transport.h b/include/tap_protocol/transport.h
index a9744eb..54b75fa 100644
--- a/include/tap_protocol/transport.h
+++ b/include/tap_protocol/transport.h
@@ -2,7 +2,7 @@
 #define TRANSPORT_H
 
 #include "tap_protocol.h"
-#include "nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 
 using json = nlohmann::json;
 
diff --git a/include/tap_protocol/utils.h b/include/tap_protocol/utils.h
index 320f247..9152748 100644
--- a/include/tap_protocol/utils.h
+++ b/include/tap_protocol/utils.h
@@ -1,9 +1,9 @@
 #ifndef UTILS_H
 #define UTILS_H
 
-#include "tap_protocol/tap_protocol.h"
-#include "tap_protocol/secp256k1_utils.h"
-#include "nlohmann/json.hpp"
+#include "tap_protocol.h"
+#include "secp256k1_utils.h"
+#include <nlohmann/json.hpp>
 
 namespace tap_protocol {
 
