diff -urN moneychanger-9999/src/moneychanger/moneychanger.pro moneychanger-9999~/src/moneychanger/moneychanger.pro
--- moneychanger-9999/src/moneychanger/moneychanger.pro 2013-12-07 01:59:15.642300910 -0600
+++ moneychanger-9999~/src/moneychanger/moneychanger.pro        2013-12-07 02:01:25.903087476 -0600
@@ -226,6 +226,9 @@
         LIBS += -ldl
 }
 
+# Temporary hack until they fix this file
+CONFIG += c++11
+QMAKE_CXXFLAGS += -std=c++11
 
 #QMAKE_CXXFLAGS_WARN_ON  = ""
 #QMAKE_CXXFLAGS_WARN_OFF = ""

